### PR TITLE
feat: support show and cancel jobs syntax

### LINKED ISSUE #8032: ddl: cancel with SQL command
After #7917, we can cancel a heavy DDL job by sending a `⌃C` signal; after #7914, we can display the progress of each running DDL job. Based on this, we can also support canceling a DDL job with a SQL command, like `ADMIN CANCEL DDL` in TiDB. Further discussion is required for the syntax.

> The `ADMIN CANCEL DDL` statement allows you to cancel a running DDL job. The job_id can be found by running [`ADMIN SHOW DDL JOBS`](https://docs.pingcap.com/tidb/stable/sql-statement-admin-show-ddl).
> https://docs.pingcap.com/tidb/stable/sql-statement-admin-cancel-ddl

### PR DESCRIPTION
**Pull request by yezizp2012**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

Resolve #8032 , support syntax to show and cancel creating streaming jobs:
```sql
dev=> show jobs;
  Id  |                     Statement                     | Progress
------+---------------------------------------------------+----------
 1010 | CREATE MATERIALIZED VIEW mv3 AS SELECT * FROM mv1 | 2.21%
 1012 | CREATE MATERIALIZED VIEW mv2 AS SELECT * FROM mv1 | 0.86%
(2 rows)

dev=> cancel jobs 1010, 1012;
  Id
------
 1012
 1010
(2 rows)
```

## Checklist

- [ ] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [ ] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [ ] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [ ] My PR needs documentation updates. (Please use the **Release note** section below to summarize the impact on users)

## Release note

If this PR includes changes that directly affect users or other significant modifications relevant to the community, kindly draft a release note to provide a concise summary of these changes. Please prioritize highlighting the impact these changes will have on users.

support syntax to show and cancel creating streaming jobs.

### CODE REVIEW

File: src/meta/src/stream/stream_manager.rs [hunk 1]
```diff
@@ -78,7 +78,8 @@ impl CreateStreamingJobContext {
 
 pub enum CreatingState {
     Failed { reason: MetaError },
-    Canceling,
+    // sender is used to notify the canceling result.
+    Canceling(oneshot::Sender<()>),
```
Comment (BugenZhao) [inline src/meta/src/stream/stream_manager.rs:82]: Suggest making this a named field.

File: src/meta/src/stream/stream_manager.rs [hunk 2]
```diff
@@ -540,9 +545,24 @@ where
         Ok(())
     }
 
-    pub async fn cancel_streaming_jobs(&self, table_ids: Vec<TableId>) {
+    pub async fn cancel_streaming_jobs(&self, table_ids: Vec<TableId>) -> Vec<TableId> {
```
Comment (BugenZhao) [inline src/meta/src/stream/stream_manager.rs:548]: May leave some docs to show what we'll return.

**Review by BugenZhao — APPROVED** [review]
LGTM!!

**Comment by yezizp2012** [comment]
@neverchanje suggested that not to introduce a new job concept but to cancel creating streaming jobs with existed command like `DROP MATERIALIZED VIEW xx`. It makes sense to me, but in the current implementation we don't have streaming jobs exposed and persisted on the frontend and in the meta during creation, which would make the implementation a bit cumbersome, and having the `DROP MATERIALIZED VIEW` semantics ambiguous, especially since we also support the `IF NOT EXISTS` keywords.
But in this future, when we support #8051 and expose it in frontend, using `DROP` syntax will be more logical and natural. For now let's support the job syntax firstly to let user can cancel it first.

**Comment by BugenZhao** [comment]
What about simply calling it `DDL` or `DDL JOB`?

**Pull request merged**